server/vulnerabilities/nvd/sync/cve_syncer.go

```diff
@@ -0,0 +1,727 @@
+						Vulnerable:            cpeMatch.Vulnerable,
+					})
+				}
+				nodes = append(nodes, &schema.NVDCVEFeedJSON10DefNode{
+					CPEMatch: cpeMatches,
+					Children: []*schema.NVDCVEFeedJSON10DefNode{},
+					Negate:   *node.Negate,
+					Operator: string(node.Operator),
+				})
+			}
+		}
+	}
+
+	var baseMetricV2 *schema.NVDCVEFeedJSON10DefImpactBaseMetricV2
+	for _, cvssMetricV2 := range cve.CVE.Metrics.CVSSMetricV2 {
+		if cvssMetricV2.Type != "Primary" {
+			continue
+		}
+		baseMetricV2 = &schema.NVDCVEFeedJSON10DefImpactBaseMetricV2{
+			AcInsufInfo: *cvssMetricV2.ACInsufInfo,
+			CVSSV2: &schema.CVSSV20{
+				AccessComplexity:           derefPtr(cvssMetricV2.CVSSData.AccessComplexity),
+				AccessVector:               derefPtr(cvssMetricV2.CVSSData.AccessVector),
+				Authentication:             derefPtr(cvssMetricV2.CVSSData.Authentication),
+				AvailabilityImpact:         derefPtr(cvssMetricV2.CVSSData.AvailabilityImpact),
+				AvailabilityRequirement:    derefPtr(cvssMetricV2.CVSSData.AvailabilityRequirement),
+				BaseScore:                  cvssMetricV2.CVSSData.BaseScore,
+				CollateralDamagePotential:  derefPtr(cvssMetricV2.CVSSData.CollateralDamagePotential),
+				ConfidentialityImpact:      derefPtr(cvssMetricV2.CVSSData.ConfidentialityImpact),
+				ConfidentialityRequirement: derefPtr(cvssMetricV2.CVSSData.ConfidentialityRequirement),
+				EnvironmentalScore:         derefPtr(cvssMetricV2.CVSSData.EnvironmentalScore),
+				Exploitability:             derefPtr(cvssMetricV2.CVSSData.Exploitability),
+				IntegrityImpact:            derefPtr(cvssMetricV2.CVSSData.IntegrityImpact),
+				IntegrityRequirement:       derefPtr(cvssMetricV2.CVSSData.IntegrityRequirement),
+				RemediationLevel:           derefPtr(cvssMetricV2.CVSSData.RemediationLevel),
+				ReportConfidence:           derefPtr(cvssMetricV2.CVSSData.ReportConfidence),
+				TargetDistribution:         derefPtr(cvssMetricV2.CVSSData.TargetDistribution),
+				TemporalScore:              derefPtr(cvssMetricV2.CVSSData.TemporalScore),
+				VectorString:               cvssMetricV2.CVSSData.VectorString,
+				Version:                    cvssMetricV2.CVSSData.Version,
+			},
+			ExploitabilityScore:     derefPtr((*float64)(cvssMetricV2.ExploitabilityScore)),
+			ImpactScore:             derefPtr((*float64)(cvssMetricV2.ImpactScore)),
+			ObtainAllPrivilege:      derefPtr(cvssMetricV2.ObtainAllPrivilege),
+			ObtainOtherPrivilege:    derefPtr(cvssMetricV2.ObtainOtherPrivilege),
+			ObtainUserPrivilege:     derefPtr(cvssMetricV2.ObtainUserPrivilege),
+			Severity:                derefPtr(cvssMetricV2.BaseSeverity),
+			UserInteractionRequired: derefPtr(cvssMetricV2.UserInteractionRequired),
+		}
+	}
+
+	var baseMetricV3 *schema.NVDCVEFeedJSON10DefImpactBaseMetricV3
+	for _, cvssMetricV30 := range cve.CVE.Metrics.CVSSMetricV30 {
+		if cvssMetricV30.Type != "Primary" {
+			continue
+		}
+		baseMetricV3 = &schema.NVDCVEFeedJSON10DefImpactBaseMetricV3{
```

Comment:
The test doesn't appear to cover this case. Can you add an entry to the test that covers it?

Reply:
I will write tests on Wed to increase coverage.